src/types/closure.md

```diff
@@ -316,65 +316,69 @@ If a closure captures a field of a composite types such as structs, tuples, and
         * If Place contains a deref of a raw pointer:
             * Let Place1 = Place truncated just before the deref
-            * Return (Mode, Place1)
+            * Return (Place1, Mode)
         * If Mode is `ref *` and the place contains a field of a packed struct:
             * Let Place1 = Place truncated just before the field
-            * Return (Mode, Place1)
+            * Return (Place1, Mode)
         * Else
-            * Return (Mode, Place1)
-    * `move_xform(Mode, Place) -> (Mode, Place)` (For move closures)
+            * Return (Place, Mode)
+    * `move_xform(Place, Mode) -> (Place, Mode)` (For move closures)
         * "Take ownership if data being accessed is owned by the variable used to access it (or if closure attempts to move data that it doesn't own)."
         * "When taking ownership, only capture data found on the stack."
         * "Otherwise, reborrow the reference."
         * If Mode is `ref mut` and the place contains a deref of an `&mut`:
-            * Return (Mode, Place)
+            * Return (Place, Mode)
         * Else if Mode is `ref *` and the place contains a deref of an `&`:
-            * Return (Mode, Place)
+            * Return (Place, Mode)
         * Else if place contains a deref:
             * Let Place1 = Place truncated just before the deref
-            * Return (ByValue, Place1)
+            * Return (Place1, ByValue)
         * Else:
-            * Return (ByValue, Place)
-    * `ref_xform(Mode, Place) -> (Mode, Place)` (for ref closures)
+            * Return (Place, ByValue)
+    * `ref_xform(Place, Mode) -> (Place, Mode)` (for ref closures)
         * "If taking ownership of data, only move data from enclosing stack frame."
         * Generate C' by mapping each (Mode, Place) in C
             * If Mode is ByValue and place contains a deref:
                 * Let Place1 = Place truncated just before the deref
-                * Return (ByValue, Place1)
+                * Return (Place1, ByValue)
             * Else:
-                * Return (Mode, Place)
-    * `ref_opt(Mode, Place) -> (Mode, Place)` (for ref closures)
+                * Return (Place, Mode)
+    * `ref_opt(Place, Mode) -> (Place, Mode)`
         * "Optimization: borrow the ref, not data owned by ref."
-        * If Place contains a deref of an `&`...
-            * ...or something
+        * Disjoint capture over immutable reference doesn't add too much value because the fields can either be borrowed immutably or copied.
+        * Edge case: Field that is accessed via the referece lives longer than the reference.
+            * Resolution: Only consider the last Deref
+        * If Place is (Base, Projections), where Projections is a list of size N.
+            * For all `i, 0 <= i < N`, Projections[i] != Deref
+                * Return (Place, Mode)
+            * If `l, 0 <= l < N` is the last/rightmost Deref Projection i.e. for any `i, l < i < N` Projection[i] != Deref,
```

Comment:
Not sure if this became too mathematical  😅

Reply:
seems good to me :)